discovery-provider/src/queries/get_saves.py

```diff
@@ -1,11 +1,54 @@
 from src import exceptions
 from src.models import Playlist, Save, SaveType, Track
-from src.queries.query_helpers import paginate_query
+from src.models.models import RepostType
+from src.queries.query_helpers import (
+    add_users_to_tracks,
+    get_users_by_id,
+    get_users_ids,
+    paginate_query,
+    populate_playlist_metadata,
+    populate_track_metadata,
+)
 from src.utils import helpers
 from src.utils.db_session import get_db_read_replica
 
 
-def get_saves(save_type, user_id):
+def populate_save_items(session, saves, save_items, save_type, current_user_id):
```

Comment:
can type if you wanted

Reply:
oh also, I noticed you're using the str track, playlist, album - if you wanted can use SaveTypeEnum  - idk if it'd help with types later on when using the function.